lib/logstash/plugin_mixins/ecs_compatibility_support/target_check.rb

```diff
@@ -0,0 +1,56 @@
+# encoding: utf-8
+
+require_relative '../ecs_compatibility_support'
+
+module LogStash
+  module PluginMixins
+    module ECSCompatibilitySupport
+      # A target option check that can be included into any `LogStash::Plugin`.
+      #
+      # @see ECSCompatibilitySupport()
+      module TargetCheck
+        ##
+        # @api internal (use: `LogStash::Plugin::include`)
+        # @param base [Class]: a class that inherits `LogStash::Plugin`, typically one
+        #                      descending from one of the four plugin base classes
+        #                      (e.g., `LogStash::Inputs::Base`)
+        # @return [void]
+        def self.included(base)
+          fail(ArgumentError, "`#{base}` must inherit LogStash::Plugin") unless base < LogStash::Plugin
+          fail(ArgumentError, "`#{base}` must include #{ECSCompatibilitySupport}") unless base < ECSCompatibilitySupport
+          base.prepend(RegisterDecorator)
+        end
+
+        TARGET_NOT_SET_MESSAGE = ("ECS compatibility is enabled but no `target` option was specified, " +
+            "it is recommended to set the option to avoid potential schema conflicts (if your data is ECS compliant " +
+            "or non-conflicting feel free to ignore this message)").freeze
```

Comment:
I'd like for this message to include "Elastic Common Schema" _in addition to_ the ECS abbreviaton. Especially as we move to having ECS on-by-default in LS8, there is a likelihood that new users won't have the background necessary from acronyms alone.

```suggestion
        TARGET_NOT_SET_MESSAGE = ("ECS compatibility is enabled but `target` option was not specified. " +
            "This may cause fields to be set at the top-level of the event where they are likely to clash with the Elastic Common Schema. " +
            "It is recommended to set the `target` option to avoid potential schema conflicts (if your data is ECS compliant " +
            "or non-conflicting, feel free to ignore this message)").freeze
```